homeassistant/components/vlc_telnet/media_player.py

```diff
@@ -148,80 +140,43 @@ async def async_update(self) -> None:
-                self._media_position = vlc_position
+            if vlc_position != self._attr_media_position:
+                self._attr_media_position_updated_at = dt_util.utcnow()
+                self._attr_media_position = vlc_position
 
         info = await self._vlc.info()
         data = info.data
         LOGGER.debug("Info data: %s", data)
 
         self._attr_media_album_name = data.get("data", {}).get("album")
-        self._media_artist = data.get("data", {}).get("artist")
-        self._media_title = data.get("data", {}).get("title")
+        self._attr_media_artist = data.get("data", {}).get("artist")
+        self._attr_media_title = data.get("data", {}).get("title")
         now_playing = data.get("data", {}).get("now_playing")
 
         # Many radio streams put artist/title/album in now_playing and title is the station name.
         if now_playing:
-            if not self._media_artist:
-                self._media_artist = self._media_title
-            self._media_title = now_playing
+            if not self._attr_media_artist:
+                self._attr_media_artist = self._attr_media_title
+            self._attr_media_title = now_playing
 
-        if self._media_title:
+        if self._attr_media_title:
             return
 
         # Fall back to filename.
         if data_info := data.get("data"):
-            self._media_title = data_info["filename"]
+            self._attr_media_title = data_info["filename"]
 
             # Strip out auth signatures if streaming local media
-            if self._media_title and (pos := self._media_title.find("?authSig=")) != -1:
-                self._media_title = self._media_title[:pos]
-
-    @property
-    def available(self) -> bool:
-        """Return True if entity is available."""
-        return self._available
-
-    @property
-    def volume_level(self) -> float | None:
-        """Volume level of the media player (0..1)."""
-        return self._volume
-
-    @property
-    def is_volume_muted(self) -> bool | None:
-        """Boolean if volume is currently muted."""
-        return self._muted
-
-    @property
-    def media_duration(self) -> int | None:
-        """Duration of current playing media in seconds."""
-        return self._media_duration
-
-    @property
-    def media_position(self) -> int | None:
-        """Position of current playing media in seconds."""
-        return self._media_position
-
-    @property
-    def media_position_updated_at(self) -> datetime | None:
-        """When was the position of the current playing media valid."""
-        return self._media_position_updated_at
-
-    @property
-    def media_title(self) -> str | None:
-        """Title of current playing media."""
-        return self._media_title
-
-    @property
-    def media_artist(self) -> str | None:
-        """Artist of current playing media, music track only."""
-        return self._media_artist
+            if (
+                self._attr_media_title
```

Comment:
```suggestion
                self.media_title
```

Reply:
Please use a walrus.

Reply:
I did not really see where you wanted the walrus